superset-frontend/src/explore/reducers/exploreReducer.js

```diff
@@ -148,9 +167,18 @@ export default function exploreReducer(state = {}, action) {
         ...getControlStateFromControlConfig(controlConfig, state, action.value),
       };
 
+      const column_config = {
+        ...state.controls.column_config,
+        ...(new_column_config && { value: new_column_config }),
+      };
+
       const newState = {
         ...state,
-        controls: { ...state.controls, [action.controlName]: control },
+        controls: {
+          ...state.controls,
+          [action.controlName]: control,
```

Comment:
you can use `controlName` directly here

Reply:
Correct, we can use it. 
Actually this change was made by vsCode auto lint, which means original code was like it. 🤣